src/provider/neoOpenSearchWorkTransformer.js

```diff
@@ -0,0 +1,152 @@
+    //            If the fields are empty, get all info (collection, relations,
+    //            dkabm, briefDisplay).
+    // subnote: when querying a pid using the search-method us a cql like this:
+    //          rec.id='pid'
+    //
+    let requestParams = {
+      action: 'getObject',
+      identifier: pid,
+      agency: neoContext.agency,
+      profile: neoContext.profile,
+      objectFormat: ['dkabm', 'briefDisplay'],
+      outputType: 'json'
+    };
+    let state = {}; // no state needed in this transformer.
+    return {transformedRequest: requestParams, state: state};
+  }
+
+  // function Z(value, key) {
+  //  let y = {};
+  //  console.log('Y: ' + key);
+  //  _.forEach(value, function (z, k) {
+  //    let x = {};
+  //    if (_.has(z, '$') && _.has(z, '@')) {
+  //      x.ns = z['@'];
+  //      x.value = z.$;
+  //      if (_.has(z, '@type')) {
+  //        x.type = z['@type'].$;
+  //      }
+  //    }
+  //    console.log(JSON.stringify(x, null, 0));
+  //  });
+  //  return y;
+  // }
+
+  function T(lookup, result) {
+
+    return function (value, key) {
+      let a = [];
+      _.forEach(value, function (z, k) { // eslint-disable-line
+        let x = {key: key};
+        if (_.has(z, '$') && _.has(z, '@')) {
+          x.ns = z['@'];
+          x.value = z.$;
+          if (_.has(z, '@type')) {
+            x.type = z['@type'].$;
+          }
+        }
+        // console.log(JSON.stringify(x, null, 0));
+        if (x.value) {
+          a.push(x);
+        }
+      });
+
+      a.map(X => {
+        let k = X.ns + ':' + key;
+        if (X.type) {
+          k += '/' + X.type;
+        }
+        if (result[k]) {
+          result[k].push(X.value);
+        } else { // eslint-disable-line brace-style
+          result[k] = [X.value];
+        }
+      });
+    };
+  }
+
+  function responseTransform(response, context, state) { // eslint-disable-line no-unused-vars
+    // TODO: check that all the below properties are valid.
+    let namespaces = response['@namespaces'];
+    let searchResult = response.searchResponse.result.searchResult;
+    let record = searchResult[0].collection.object[0].record;
+
+    let lookup = {namespaces: namespaces};
+    let result = {};
+    _.forOwn(record, T(lookup, result));
+
+    let data = {};
+    data.warning = 'DO NOT USE THIS RESULT. It is invalid';
+    _.extend(data, result);
+
+    // console.log("RES: " + JSON.stringify(data, null, 4));
+
+    data.title = 'HELLO';
+    return {statusCode: 200, data: [data]};
+  }
+
+  function OSWorkFunc(context) {
+    let neoContext = context.libdata.config.provider.services.opensearch;
+    // console.log("Context: " + JSON.stringify(neoContext, null, 4));
+    let method = ''; // method is empty in this request. It is given as 'action' in the parameter-object.
+    let config = {
+      endpoint: neoContext.wsdl.slice(0, -5) // hack! the context gives us an url with '?wsdl'. This i part is removed.
```

Comment:
I am totally aware of that. The purpuse of this pull is just to get something partially working integrated into the master, for others start using. It is in no way complete, just an initial stab at some almost correct functionality.